Add connection_total metric and deprecate connection_error_total (#218)
strimzi/strimzi-canary: README.md

| `consumer_timeout_join_group_total`     | The total number of consumers not joining the group within the timeout            |
| `consumer_refresh_metadata_error_total` | Total number of errors while refreshing consumer metadata                         |
| `records_consumed_latency`              | Records end-to-end latency in milliseconds                                        |

> | `connection_error_total`                | *Deprecated - use connection_total* Total number of errors while checking the connection to Kafka brokers|

Reviewer: ```suggestion
| `connection_error_total`                | (Deprecated - `use connection_total`) Total number of errors while checking the connection to Kafka brokers|
```

Author: done (with the backticks around connection_total excluding the use)

> | `connection_total`                      | Total of outcomes while checking the connection to Kafka brokers                  |

Reviewer: ```suggestion
| `connection_total`                      | Total number of connections to Kafka brokers                  |
```

Reviewer: I wasn't sure what _outcomes_ meant here. Suggestion assumes we're just returning total number of connections.  We could add succeeded or failed too: _Total number of (failed or successful) connections to Kafka brokers_
